src/main/java/duke/Task.java

```diff
@@ -0,0 +1,101 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represent a task to complete
+ */
+public class Task {
+    boolean isDone;
+    protected final String task;
+    protected final String type;
+    protected final String time;
+    protected LocalDate date;
+    protected static DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MMM d yyyy");
+
+    /**
+     * Constructor of task
+     * @param task Description of the task
+     * @param type Type of the task
+     */
+    Task(String task, String type) {
+        this.isDone = false;
+        this.task = task;
+        this.type = type;
+        this.date = LocalDate.now();
+        this.time = this.date.toString();
+    }
+
+    /**
+     * Constructor of task
+     * @param task Description of the task
+     * @param type Type of the task
+     * @param time time of the task
+     */
+    Task(String task, String type, String time) {
+        this.isDone = false;
+        this.task = task;
+        this.type = type;
+        this.time = time;
+        String timeArr[] = time.split(" ");
+        this.date = LocalDate.parse(timeArr[1]);
+    }
+
+    /**
+     * Check if the description of task contains certain word
+     * @param word The word wish to check
+     * @return Boolean of whether the word is contained in task
+     */
+    boolean containsWord(String word) {
+        return this.task.contains(word);
+    }
+    /**
+     * Mark the task as done
+     */
+    void mark() {
+        this.isDone = true;
+    }
+
+    /**
+     * Unmark the task as not done
+     */
+    void unMark() {
```

Comment:
Should this be changed to adhere to coding standards?
```suggestion
    void setUnmark() {
```